www/include/common/javascript/datepicker/localizedDatepicker.js

```diff
@@ -132,58 +130,100 @@ function setUserFormat() {
     $(".datepicker").first().on('change', function (e) {
-        updateEndTime();
+        updateDateAndTime();
     });
     $(".timepicker").first().on('change', function (e) {
-        updateEndTime();
+        // On change, the first click is taken in account. but the second click will update the end timepicker,
+        // as the focus is lost on the first timepicker when updating the end timepicker value.
+        // So the timepicker popin needs to be hidden while the end value is modified.
+        $(".ui-timepicker.ui-widget").hide();
+        updateDateAndTime();
     });
+
+    // End value of datepicker and timepicker selector
     $(".datepicker").last().on('change', function (e) {
-        updateStartTime();
+        updateEndTime();
     });
     $(".timepicker").last().on('change', function (e) {
-        updateStartTime();
+        // In this case, we should not update the start time (could be in the past), but only the end time.
+        // Update the end time to a consistent value (with the chosen duration) and alert the user about it.
+        checkEndTime();
     });
 }
 
+/**
+ * Turn the event Off to avoid infinite loop.
+ */
 function turnOffEvents() {
     $(".datepicker").off('change');
     $(".timepicker").off('change');
 }
 
-function updateEndTime() {
+/**
+ * Update the end datepicker and timepicker according to the start values.
+ */
+function updateDateAndTime() {
     let start = moment($('[name="alternativeDateStart"]').val()
-        + ' ' +  $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
     let end = moment($('[name="alternativeDateEnd"]').val()
-        + ' ' +  $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
 
     if (start.isAfter(end) || start.isSame(end)) {
         turnOffEvents();
         start.add($('#duration').val(), $('#duration_scale').val());
         $(".datepicker").last().datepicker("setDate", start.format($(".datepicker").last().datepicker(
             "option",
             "dateFormat"
-        ).toUpperCase().replace(/Y/g,'YY')));
+        ).toUpperCase().replace(/Y/g, 'YY')));
         $(".timepicker").last().timepicker("setTime", start.format("HH:mm"));
         turnOnEvents();
     }
 }
 
-function updateStartTime() {
+/**
+ * Used for the end DATEPICKER date.
+ * Update the end timepicker according to the start values.
+ */
+function updateEndTime() {
     let start = moment($('[name="alternativeDateStart"]').val()
-        + ' ' +  $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
     let end = moment($('[name="alternativeDateEnd"]').val()
-        + ' ' +  $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
 
     if (start.isAfter(end) || start.isSame(end)) {
```

Comment:
you could use `isSameOrAfter` : https://momentjscom.readthedocs.io/en/latest/moment/05-query/05-is-same-or-after/